src/Sylius/Bundle/AdminBundle/Menu/ProductUpdateMenuBuilder.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Menu;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Sylius\Bundle\AdminBundle\Event\ProductMenuBuilderEvent;
+use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class ProductUpdateMenuBuilder
+{
+    public const EVENT_NAME = 'sylius.menu.admin.product.update';
+
+    /** @var FactoryInterface */
+    private $factory;
+
+    /** @var EventDispatcherInterface */
+    private $eventDispatcher;
+
+    public function __construct(FactoryInterface $factory, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->factory = $factory;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function createMenu(array $options): ItemInterface
+    {
+        $menu = $this->factory->createItem('root');
+
+        if (!isset($options['product'])) {
+            return $menu;
+        }
+
+        $product = $options['product'];
+        if (!$product instanceof ProductInterface) {
```

Comment:
Can we also skip this logic for simple products?

Reply:
For simple products, there is possibility to create variant (the same behaviour as on index page)